packages/flutter/test/widgets/scrollable_semantics_test.dart

```diff
@@ -399,6 +387,135 @@ void main() {
+    tester.binding.pipelineOwner.semanticsOwner.performAction(firstContainerId, SemanticsAction.showOnScreen);
+    await tester.pumpAndSettle();
+
+    expect(scrollController.offset, 0.0);
+
+    semantics.dispose();
+  });
+
+  testWidgets('showOnScreen brings item below trailing edge to trailing edge', (WidgetTester tester) async {
+    semantics = new SemanticsTester(tester); // enables semantics tree generation
+
+    const double kItemHeight = 100.0;
+    final List<Widget> children = new List<Widget>.generate(10, (int i) {
+      return new MergeSemantics(
+        child: new Container(
+          height: kItemHeight,
+          child: new Text('container $i'),
+        ),
+      );
+    });
+
+    final ScrollController scrollController = new ScrollController(
+      initialScrollOffset: kItemHeight / 2,
+    );
+
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new Center(
+          child: new Container(
+            height: 2 * kItemHeight,
+            child: new ListView(
+              controller: scrollController,
+              children: children,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(scrollController.offset, kItemHeight / 2);
+
+    final int firstContainerId = tester.renderObject(find.byWidget(children[2])).debugSemantics.id;
+    tester.binding.pipelineOwner.semanticsOwner.performAction(firstContainerId, SemanticsAction.showOnScreen);
+    await tester.pumpAndSettle();
+
+    expect(scrollController.offset, kItemHeight);
+
+    semantics.dispose();
+  });
+
+  testWidgets('showOnScreen does not change position of items already fully on-screen', (WidgetTester tester) async {
+    semantics = new SemanticsTester(tester); // enables semantics tree generation
+
+    const double kItemHeight = 100.0;
+    final List<Widget> children = new List<Widget>.generate(10, (int i) {
+      return new MergeSemantics(
+        child: new Container(
+          height: kItemHeight,
+          child: new Text('container $i'),
+        ),
+      );
+    });
+
+    final ScrollController scrollController = new ScrollController(
+      initialScrollOffset: kItemHeight / 2,
+    );
+
+    await tester.pumpWidget(
+      new Directionality(
+        textDirection: TextDirection.ltr,
+        child: new Center(
+          child: new Container(
+            height: 2 * kItemHeight,
+            child: new ListView(
+              controller: scrollController,
+              children: children,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(scrollController.offset, kItemHeight / 2);
+
+    final int firstContainerId = tester.renderObject(find.byWidget(children[1])).debugSemantics.id;
+    tester.binding.pipelineOwner.semanticsOwner.performAction(firstContainerId, SemanticsAction.showOnScreen);
+    await tester.pumpAndSettle();
+
+    expect(scrollController.offset, kItemHeight / 2);
+
+    semantics.dispose();
+  });
```

Comment:
we should test that this works with negative-scroll-offset children too

Reply:
Good idea. Added.